Makefile

```diff
@@ -472,7 +472,7 @@ source-dist:
 # Make tarball with Julia code plus all dependencies
 full-source-dist: light-source-dist.tmp
 	# Get all the dependencies downloaded
-	@$(MAKE) -C deps getall NO_GIT=1
+	@$(MAKE) -C deps getall DEPS_GIT=0 USE_BINARYBUILDER=0
```

Comment:
We specify `USE_BINARYBUILDER=1` and `USE_BINARYBUILDER=0` to generate two different kinds of `full-source-dist` tarballs.  You shouldn't set it here unconditionally.

Reply:
X-ref: https://github.com/JuliaCI/julia-buildbot/blob/138f0c5181bcbf1c1a802674c44d19ab18d8db1a/master/doctest.py#L98-L132

Reply:
I think you misunderstand the target definitions here. The `make -C get` target would be what is used to grab all of the binary files needed to build on this machine. The `getall` target grabs all of the source files needed to build on any machine. The `getall` target therefore logically cannot be used with USE_BINARYBUILDER=1. I could add `$(MAKE) -f contrib/refresh_checksums.mk` here though, conditional on `USE_BINARYBUILDER=1`, if you like.

Reply:
We have customers that are using things like `make -C deps getall USE_BINARYBUILDER=1 USE_BINARYBUILDER_XXX=0` to generate a tarball that they can deploy to CI and patch with their own patches to deps and such without having to rebuild LLVM, for instance.  What would you suggest they use instead?

Reply:
They should use `get` (or perhaps `extract`), which is the first stage of the build pipeline (or the second, which also supports adding custom patches after). The getall target is not part of the build pipeline, but is instead part of the uninstall / cleanall set, which are a list of all possible targets, not just those that currently need to be built locally.

Reply:
okay; once this is merged let's double-check that the generated tarballs look right.